### ISSUE TITLE
window.matchMedia is not a function errors after ugprade to 41.4.1

### ISSUE THREAD

**Issue by alexander-schranz** [ckeditor/ckeditor5#16368]
## 📝 Provide detailed reproduction steps (if any)

1. Run CKEditor in a CLI Environment without access to `window` like Jest (in our Jest 24)

### ✔️ Expected result

_What is the expected result of the above steps?_

### ❌ Actual result

```
 TypeError: window.matchMedia is not a function

      2 | import React from 'react';
      3 | import log from 'loglevel';
    > 4 | import AlignmentPlugin from '@ckeditor/ckeditor5-alignment/src/alignment';
        | ^
      5 | import BoldPlugin from '@ckeditor/ckeditor5-basic-styles/src/bold';
      6 | import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
      7 | import EssentialsPlugin from '@ckeditor/ckeditor5-essentials/src/essentials';

      at isMediaForcedColors (node_modules/@ckeditor/ckeditor5-utils/src/env.js:131:19)
      at Object.<anonymous> (node_modules/@ckeditor/ckeditor5-utils/src/env.js:34:26)
      at Object.<anonymous> (node_modules/@ckeditor/ckeditor5-utils/src/index.js:8:1)
      at Object.<anonymous> (node_modules/@ckeditor/ckeditor5-core/src/plugin.js:9:1)
      at Object.<anonymous> (node_modules/@ckeditor/ckeditor5-core/src/index.js:8:1)
      at Object.<anonymous> (node_modules/ckeditor5/src/core.js:8:1)
      at Object.<anonymous> (node_modules/@ckeditor/ckeditor5-alignment/src/alignment.js:8:1)
      at Object.<anonymous> (src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/CKEditor5.js:4:1)
      at Object.<anonymous> (src/Sulu/Bundle/AdminBundle/Resources/js/containers/CKEditor5/index.js:2:1)
      at Object.<anonymous> (src/Sulu/Bundle/AdminBundle/Resources/js/containers/index.js:2:1)
      at Object.<anonymous> (src/Sulu/Bundle/PageBundle/Resources/js/containers/TeaserSelection/tests/TeaserSelection.test.js:5:1)
```

### ❓ Possible solution

For server side renderers and cli best rendering it would be best if the `window.matchMedia` is only access when windowvariable is available.

## 📃 Other details

* Browser: none
* OS: Mac and Linux
* First affected CKEditor version: think 41.4.1 (CI did run yesterday but today not)
* Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Thanks for reporting.

It's indeed a regression. We should not block importing the entire package in an env like Jest.

The problem is here:

https://github.com/ckeditor/ckeditor5/blob/78174de1c2ddf9f53c76aef903e73f05f877b0f2/packages/ckeditor5-utils/src/env.ts#L112-L113

And it should probably be a getter like this one:

[https://github.com/ckeditor/ckeditor5/blob/78174de1c2ddf9f53c76aef903e73f05f877b0f2/packages/ckeditor5-utils/src/env.ts#L114-L116](https://github.com/ckeditor/ckeditor5/blob/78174de1c2ddf9f53c76aef903e73f05f877b0f2/packages/ckeditor5-utils/src/env.ts#L114-L116)

**Comment by alexander-schranz** [comment]
@Reinmar Thank you for the fast response!

**Comment by Reinmar** [comment]
Two questions:

*   Will the change to getter resolve this specific problem?
*   Didn't we (by coincident) introduce more issues like this one, preventing importing a specific module in a server-side setting?

Bonus thing: We should such scenarios this to our E2E tests (and cover as many packages as actually work today). cc @Mgsy @dufipl 

@alexander-schranz Do you know about any of our packages which would not be importable in Jest already?

**Comment by Reinmar** [comment]
What about before v41.4.1? Did you have to work around any similar issue in any other package?

**Comment by alexander-schranz** [comment]
@Reinmar as a workaround I found out I could create in Jest a instance of `window.matchMedia` this way but requires to change the setup script of jest that it is available globally: https://jestjs.io/docs/29.4/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

**Comment by scofalik** [comment]
@alexander-schranz I'd like to make sure that this is the only error we introduced in this release. Could you please confirm if after mocking `window.matchMedia` you are able to run tests in Jest successfully?

**Comment by alexander-schranz** [comment]
@scofalik yes the `window.matchMedia` was in our project the only case which did fail.

**Comment by scofalik** [comment]
This issue should be closed on `release` branch.

**Comment by Reinmar** [comment]
Released in v41.4.2: https://github.com/ckeditor/ckeditor5/releases/tag/v41.4.2
